Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -141,4 +153,25 @@ fileprivate extension TrialOrIntroPriceEligibilityChecker {
         }
     }
 
+    @available(iOS 11.2, macOS 10.13.2, macCatalyst 13.0, tvOS 11.2, watchOS 6.2, *)
+    func productsWithIntroOffers(productIdentifiers: [String]) -> [String: IntroEligibility] {
+        var productIdsToIntroEligibleStatus: [String: IntroEligibility] = [:]
+        let group = DispatchGroup()
```

Comment:
Seems weird to make this blocking.
Considering that the public method `Purchases.checkTrialOrIntroductoryPriceEligibility` is already "async" (completion block), maybe use the same pattern here?

Reply:
I was thinking the same. My initial thought was it makes the logic easier if we return a value instead of completion, but you're right.

Reply:
Done ✅